packages/engine-multi/src/worker/thread/helpers.ts

```diff
@@ -66,17 +65,23 @@ export const createLoggers = (
 // Execute wrapper function
 export const execute = async (
   workflowId: string,
-  executeFn: () => Promise<any> | undefined
+  executeFn: () => Promise<any> | undefined,
+  publishFn = publish
 ) => {
   const handleError = (err: any) => {
-    publish(workerEvents.ERROR, {
+    publishFn(workerEvents.ERROR, {
       // @ts-ignore
       workflowId,
       // Map the error out of the thread in a serializable format
       error: serializeError(err),
-      stack: err?.stack
+      stack: err?.stack,
       // TODO job id maybe
     });
+
+    // Explicitly send a reject task error, to ensure the worker closes down
+    publish(workerEvents.ENGINE_REJECT_TASK, {
```

Comment:
This is the actual fix and this is what WASN'T happening before.

It's not enough to tell the worker we've errored. We need to send an event to the parent process to tell it that we're dead.

I'm a little concerned about duplicating error messages here and so there's a little bit of gnarly code on the reporting side to handle that.